cmd/exec.go

```diff
@@ -306,3 +347,94 @@ func isStaleFile(file os.FileInfo) bool {
 
 	return time.Now().Sub(file.ModTime()) > timeLimit
 }
+
+func gzipLogData(logData string) *bytes.Buffer {
+	var b bytes.Buffer
+	if len(logData) < 1 {
+		return &b
+	}
+
+	gz := gzip.NewWriter(&b)
+	if _, err := gz.Write([]byte(logData)); err != nil {
+		log("error writing gzip")
+		return nil
+	}
+	if err := gz.Close(); err != nil {
+		log("error closing gzip")
+		return nil
+	}
+	return &b
+}
+
+func getPresignedUrl(postBody []byte) ([]byte, error) {
+	url := "https://cronitor.io/api/logs/presign"
+
+	client := &http.Client{Timeout: 120 * time.Second}
+	request, err := http.NewRequest("POST", url, strings.NewReader(string(postBody)))
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create request for URL presign")
+	}
+	request.SetBasicAuth(apiKey, "")
+	request.Header.Add("Content-Type", "application/json")
+	response, err := client.Do(request)
+	if err != nil {
+		return nil, errors.Wrap(err, "error requesting presigned url")
+	}
+	if response.StatusCode != 200 && response.StatusCode != 201 {
+		return nil, fmt.Errorf("error response code %d returned", response.StatusCode)
+	}
+
+	contents, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+	defer response.Body.Close()
+	response.Body = ioutil.NopCloser(bytes.NewBuffer(contents))
+	return contents, nil
+}
+
+func shipLogData(jobKey string, seriesID string, outputLogs string) ([]byte, error) {
```

Comment:
Small but change any references to "job" to say "monitor" for consistency with this code base, otherwise we're just going to confuse ourselves later

Reply:
`"job_key"` is coded as a parameter in the POST body, but other than that I can update it